src/GPS.cpp

```diff
@@ -0,0 +1,110 @@
+#include "GPS.hpp"
+#include <nmea0183/GPS.hpp>
+
+using namespace nmea0183;
+using namespace marnav;
+using namespace std;
+using namespace gps_base;
+
+GPS_SOLUTION_TYPES GPS::getPositionType(utils::optional<nmea::mode_indicator> mode)
+{
+    switch (*mode) {
+        case nmea::mode_indicator::invalid:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case nmea::mode_indicator::autonomous:
+            return GPS_SOLUTION_TYPES::AUTONOMOUS;
+        case nmea::mode_indicator::differential:
+            return GPS_SOLUTION_TYPES::DIFFERENTIAL;
+        case nmea::mode_indicator::estimated:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case nmea::mode_indicator::manual_input:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case nmea::mode_indicator::simulated:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case nmea::mode_indicator::data_not_valid:
+            return GPS_SOLUTION_TYPES::INVALID;
+        case nmea::mode_indicator::precise:
+            return GPS_SOLUTION_TYPES::AUTONOMOUS;
+        default:
+            return GPS_SOLUTION_TYPES::INVALID;
+    }
+}
+
+base::Time GPS::buildRockTime(utils::optional<nmea::time> const& optional_time,
+    utils::optional<nmea::date> const& optional_date)
+{
+    auto date = optional_date.value();
```

Comment:
Optional as a type exists to ensure the programmer is either:

1. Fine with the code throwing an exception when a value that SHOULD be there isnt;
2. Forced to check if the value is there and treat the case when it isnt

I am pretty sure this is a case of 2. here, we dont want to throw an exception when we didnt have the time